### PR TITLE
Adding a new keyboard, TL Split Keyboard 18mm

### PR DESCRIPTION
**Pull request by satromi**
## Description

Adds support for the TL Split Keyboard 18mm .

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds support for TL Split Keyboard

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tlsplit18/config.h [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0000
+#define DEVICE_VER 0x0001
+#define MANUFACTURER satromi
+#define PRODUCT tlsplit18
+#define DESCRIPTION A splite keyboard for TRON fan
```
Comment (zvecr) [inline keyboards/tlsplit18/config.h:28]:
```suggestion
```

File: keyboards/tlsplit18/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#include QMK_KEYBOARD_H
+// #include"keymap_jp.h"
+
+// #define QWERTY 0
+// #define FUNC 1
+
+enum layer_number {
+  QWERTY = 0,
+  FUNC,
+};
```
Comment (zvecr) [inline keyboards/tlsplit18/keymaps/default/keymap.c:25]:
```suggestion
// Defines names for use in layer keycodes and the keymap
enum layer_names {
    _QWERTY,
    _FN
};
```

File: keyboards/tlsplit18/rules.mk [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+SPLIT_KEYBOARD = yes		# Split Keyboard Support
```
Comment (zvecr) [inline keyboards/tlsplit18/rules.mk:33]:
```suggestion
# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
SPLIT_KEYBOARD = yes        # Split Keyboard Support
```

File: keyboards/tlsplit18/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+   */
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+   [QWERTY] = LAYOUT(
+            KC_RBRC,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,  KC_END, KC_CAPS,                   KC_KANA, KC_LEFT,   KC_6,    KC_7,    KC_8,    KC_9,    KC_0, KC_MINS,          
+   KC_ESC,  KC_NUHS,    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T, KC_PGDN, TG_FUNC,  KC_DEL,  KC_INS,KC_LANG5, KC_DOWN,   KC_Y,    KC_U,    KC_I,    KC_O,    KC_P, KC_LBRC, KC_JYEN, 
+   KC_ZKHK, KC_LALT,    KC_A,    KC_S,    KC_D,    KC_F,    KC_G, KC_HOME, KC_LSFT,  KC_TAB, KC_PENT, KC_RSFT, KC_RGHT,   KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN, KC_QUOT,  KC_EQL, 
+            KC_LCTL,    KC_Z,    KC_X,    KC_C,    KC_V,    KC_B, KC_PGUP,  KC_SPC,                   KC_BSPC,   KC_UP,   KC_N,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,   KC_RO
+   ),
+
+   [FUNC] = LAYOUT(
```
Comment (zvecr) [inline keyboards/tlsplit18/keymaps/default/keymap.c:60]:
```suggestion
   [_FN] = LAYOUT(
```

File: keyboards/tlsplit18/config.h [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+#define LCD_DATA3_PIN    7            //< pin for 4bit data bit 3
+#define LCD_RS_PORT      LCD_PORT     //< port for RS line
+#define LCD_RS_PIN       3            //< pin  for RS line
+#define LCD_RW_PORT      LCD_PORT     //< port for RW line
+#define LCD_RW_PIN       2            //< pin  for RW line
+#define LCD_E_PORT       LCD_PORT     //< port for Enable line
+#define LCD_E_PIN        1            //< pin  for Enable line
+#endif
+*/
+
```
Comment (fauxpark) [inline keyboards/tlsplit18/config.h:251]:
```suggestion
```

**Review by drashna — COMMENTED** [review]
Because of how the keyboard matrix is shaped, it may benefit from using the eager per row debounce method.

**Pull request closed without merging**